Gemfile

```diff
@@ -6,3 +6,4 @@ gemspec
 rails_version = "#{ENV['RAILS_VERSION'] || '6.0.0'}"
 
 gem "rails", rails_version == "master" ? { github: "rails/rails" } : rails_version
+gem "roadie", "~> 4.0", require: false
```

Comment:
I can see why we want to use `roadie` for this, but I'd _really_ prefer to see if we can pursue an option that lets us avoid doing so. 

Including this kind of dependency will make it more difficult to merge our code into Rails.

Reply:
Makes sense. 

So the initial approach was just rendering the previews inside an iframe. We can go back that way maybe if it makes it easier for this move forward. 

The main downsides I saw about using iframes is that I couldn't give a proper height size to the iframe itself so that it doesn't have a scroll bar when the component previewed is bigger than the default height.

Reply:
Might we be able to use `render_inline` for this functionality?

Reply:
The `render_inline` does half of the work, because it only returns the html of the rendered component. 

The other half is about including the styles for the component. And since that lives in a stylesheet, and stylesheets linked to a DOM affect the whole DOM, we cannot just link them in the view. 

So that's why I thought about using iframes so that the component is rendered isolated from the main DOM (the one containing all examples from a component) and hence the iframes have the correct stylesheet. 

This can extend to javascript files too I think.

Reply:
What do you think?

Reply:
My initial thought would be using an iframe too. The GOV.UK components sometimes suffer from styles leaking from the "guide" into the components, and vice versa. Another project in UK Government does the same (https://design-system.service.gov.uk/components/details). It solves the sizing issue by giving each example a specific size (not sure how we'd do that here).

Reply:
My thinking here is that we should consider what the best default behavior would be. To me, it might be as "simple" as using the existing application layout for these views! What do you think about that?

Reply:
Agree, we'll go that way. 

So we'll use the application layout for the examples' views.